databases/week2/homework.sql

```diff
@@ -0,0 +1,134 @@
+
+-- Part 1
+-- Add a task with these attributes: title, description, created, updated, due_date, 
+-- status_id, user_id
+
+insert into task (title, description, created, updated, due_date, status_id, user_id)
+values ("more", "more investments", "2017-10-15 13:05:09", "2017-10-15 13:05:09","2017-10-15 13:05:09", 3, 1)
+
+-- Change the title of a task
+UPDATE task SET title = "investements" WHERE id = 1;
+-- Change a task due date
+UPDATE task SET due_date = "2017-10-15 13:05:09" WHERE id = 1;
+-- Change a task status
+SELECT title FROM task WHERE id = 1;
+UPDATE task SET status_id = 1 WHERE id = 1;
+-- Mark a task as complete
+UPDATE task SET status_id = 3, description = "completed" WHERE id = 1;
+-- Delete a task
+DELETE FROM task WHERE id = 3
+
+
+
+
+-- Part 2: School database
+-- Create a new database containing the following tables:
+-- Class: with the columns: id, name, begins (date), ends (date)
+
+CREATE TABLE `Class` (
+    `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+    `name` varchar(255) NOT NULL,
+    `begins` DATETIME,
+    `ends` DATETIME,
+    PRIMARY KEY (`id`)
+ ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+INSERT INTO `Class` (`name`, `begins`, `ends`) VALUES ('Math', '2021-09-15 13:37:42', '2021-09-15 16:37:42');
+
+
+-- Student: with the columns: id, name, email, phone, class_id (foreign key)
+
+
+ CREATE TABLE `Student` (
+    `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+    `name` varchar(255) NOT NULL,
+    `email` varchar(255),
+    `phone` int(20) ,
+    `class_id` int(20) unsigned,
+       PRIMARY KEY (`id`),
+   FOREIGN KEY (`class_id`) REFERENCES `Class`(`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+ ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+INSERT INTO `Student` (`name`, `email`, `phone`,`class_id`) VALUES ('Albert', 'albert@something.com', '40404035',1);
+
+-- If you are done with the above tasks, you can continue with these advanced tasks:
+-- Create an index on the name column of the student table.
+ CREATE INDEX idx_student_name ON `Student` (`name`);
+
+-- Add a new column to the class table named status which can only have the following values:
+--  not-started, ongoing, finished (hint: enumerations).
+
+
+ ALTER TABLE `Class` ADD `status` ENUM('not-started', 'ongoing', 'finished') DEFAULT 'not-started' NOT NULL ;
+
+
+
+Part 3: More queries
+
+
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+
```

Comment:
I think you are supposed to use other tables and values for this assignment according to the homework description.